cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/FileSetProperties.java

```diff
@@ -50,18 +55,133 @@
    */
   public static final String OUTPUT_PROPERTIES_PREFIX = "output.properties.";
 
+  /**
+   * Whether this dataset should be enabled for explore.
+   */
+  public static final String PROPERTY_EXPLORE_ENABLED = "explore.enabled";
```

Comment:
Right now we have no separate endpoints or APIs to enable it after the dataset was created. So I will go with your first suggestion.